File: Alternativas/ej15.py
# -*- coding: utf-8 -*-

#El director de una escuela está organizando un viaje de estudios, y requiere 
#determinar cuánto debe cobrar a cada alumno y cuánto debe pagar 
#a la compañía de viajes por el servicio. 
#La forma de cobrar es la siguiente: 
#    -si son 100 alumnos o más, el costo por cada alumno es de 65 euros; 
#    -de 50 a 99 alumnos, el costo es de 70 euros 
#    -de 30 a 49, de 95 euros
#    -si son menos de 30, el costo de la renta del autobús es de 4000 euros, sin importar el número de alumnos. 
#Realice un algoritmo que permita determinar el pago a la compañía de autobuses y lo que debe pagar cada alumno por el viaje.

#almacenar el número de alumnos
#si son 100 o más cada alumno paga 65 y la compañia recibe 65*alumnos
#si son entre 50 y 99 el costo es 70 y la compañia recibe 70*alumnos
#si son entre 30 y 49 el costo es 95 y la compañia recibe 95*alumnos
#si son menos de 30 el coste por alumnos es 4000/alumnos y la compañia recibe 4000

print("---[Calculo viaje escolar en autobus]---")
alumnos = int(input("Introduce el número de alumnos> ")) 
if(alumnos>=100): #Si la cantidad de alumnos es mayor o igual a 100
    costeAlumno = 65
    gananciaEmpresa = 65*alumnos
elif((alumnos>50)and(alumnos<99)): #Si la cantidad de alumnos es mayor que 50 y manor que 99
    costeAlumno = 70
    gananciaEmpresa = 70*alumnos
elif((alumnos>30)and(alumnos<49)): #Si la cantidad de alumnos es mayor a 30 y menor a 49
    costeAlumno = 95
    gananciaEmpresa = 95*alumnos
else: #Si son menos de 30 alumnos
    costeAlumno = 4000/alumnos
    gananciaEmpresa = 4000

print(f"Los alumnos deberán pagar {costeAlumno} y la empresa cobrará {gananciaEmpresa}")

File: Java/Alternativas/ej15.java
import java.util.Scanner;

/**
 * El director de una escuela está organizando un viaje de estudios, 
 * y requiere determinar cuánto debe cobrar a cada alumno y cuánto debe pagar 
 * a la compañía de viajes por el servicio. 
 * La forma de cobrar es la siguiente: 
 * 		si son 100 alumnos o más, el costo por cada alumno es de 65 euros; 
 * 		de 50 a 99 alumnos, el costo es de 70 euros, 
 * 		de 30 a 49, de 95 euros, 
 * 		y si son menos de 30, el costo de la renta del autobús es de 4000 euros, 
 * 		sin importar el número de alumnos. 
 * Realice un algoritmo que permita determinar el pago a la compañía de autobuses 
 * y lo que debe pagar cada alumno por el viaje.
 * 
 * @author Alberto Miguel Martinez Jimenez
 *
 */
/**
 * Algoritmo
 * almacenar número de alumnos
 * si alumnos >= 100 pagoPorAlumno = 65  pagoACompañia = 65*numeroAlumnos
 * si alumnos >= 50 y alumnos <= 99 pagoPorAlumno = 70  pagoACompañia = 70*numeroAlumnos
 * si alumnos >= 30 y alumnos <= 49 pagoPorAlumno = 95  pagoACompañia = 95*numeroAlumnos
 * si alumnos < 30 el costo total de autobus es de 4000 euros pago por alumno = 4000/ numeroAlumnos
 */    

public class ej15 {

	public static void main(String[] args) {
		int numeroAlumnos;
		double pagoPorAlumno = 0;
		double pagoACompannia = 0;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		System.out.println("---Calcular los costes de un viaje escolar---");
		System.out.println("Introduce el número de alumnos> ");
		numeroAlumnos = scanner.nextInt();
		if(numeroAlumnos>=100) {
			pagoPorAlumno = 65;
			pagoACompannia = 65 * numeroAlumnos;
		}else if((numeroAlumnos>=50)&&(numeroAlumnos<=99)) {
			pagoPorAlumno = 70;
			pagoACompannia = 70 * numeroAlumnos;
		}else if((numeroAlumnos>=30)&&(numeroAlumnos<=49)) {
			pagoPorAlumno = 95;
			pagoACompannia = 95 * numeroAlumnos;
		}else if(numeroAlumnos<30) {
			pagoPorAlumno = numeroAlumnos/4000;
			pagoACompannia = 4000;
		}
		
		System.out.println("Cada alumno debe pagar "+pagoPorAlumno+" y el pago a la comañia es "+pagoACompannia);
	}

}
